TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -14,57 +14,45 @@ public TutorService(ITutorDataAccess dataAccess, IUserIdentityService userIdenti
     }
     public Ad CreateAd(string subject, string title, string description)
     {
-        // return created Ad
-        // return empty with id = 0 to indicate something went wrong
-        return new Ad()
+        int? tutorId = _userIdentityService.GetUserId();
+        if (tutorId is null)
         {
-            // this is only for tests
-            Id = 1,
-            Subject = subject,
-            Title = title,
-            Description = description
-        };
+            return new Ad()
+            { Id = 0 };
+        }
+
+        return _dataAccess.CreateAd((int)tutorId, subject, title, description);
+
     }
     public ScheduleItem CreateScheduleItem(int adId, DateTime dateTime)
     {
-        // return created ScheduleItem
-        // return empty with id = 0 to indicate something went wrong
-        return new ScheduleItem()
-        {
-            // this is only for tests
-            Id = 1,
-            AdId = adId,
-            DateTime = dateTime
-        };
+        return _dataAccess.CreateScheduleItem(adId, dateTime);
     }
-    public Ad? GetAdById(int adId)
+    public Ad? GetAdById(int? adId)
     {
         // return Ad (from _dataAccess) with provided adId
         // return null if no such Ad
-        return new Ad()
+        adId = _dataAccess.GetAllAdsId();
+        if (adId is null)
         {
-            // this is only for tests
-            Id = adId,
-            Title = "Test"
-        };
+            return null;
+        }
+        return _dataAccess.GetAdById((int)adId);
     }
-    public ScheduleItem? GetScheduleItemById(int scheduleItemId)
+    public ScheduleItem? GetScheduleItemById(int? scheduleItemId)
     {
-        return new ScheduleItem()
+        scheduleItemId = _dataAccess.GetScheduleItemId();
+        if (scheduleItemId is null)
         {
-            // this is only for tests
-            Id = scheduleItemId,
-            DateTime = DateTime.Now.AddHours(1),
-        };
+            return null;
+        }
+        return _dataAccess.GetScheduleItemById(scheduleItemId);
     }
-    public List<Ad> GetUsersAds()
+    public List<Ad> GetUsersAds(int userId)
     {
```

Comment:
You added the parameter in the wrong place. By putting it as the parameter of the method you are in, you declare it as something that you will receive. In this case you will have to get it yourself and push it forward to the method that actually gets you the data you are looking for, because it will need to know who the current user is and is unable to determine it by itself.